engine/src/org/pentaho/di/trans/steps/ldapinput/LdapProtocolFactory.java

```diff
@@ -89,7 +89,7 @@ public LdapProtocolFactory( LogChannelInterface log ) {
    */
   public LdapProtocol createLdapProtocol( VariableSpace variableSpace, LdapMeta meta,
     Collection<String> binaryAttributes ) throws KettleException {
-    String connectionType = meta.getProtocol();
+    String connectionType = environmentSubstitute( meta.getProtocol() );
```

Comment:
This doesn't compile:
Wed Mar 01 13:35:56 EST 2017 - stdout - head:     [javac] /home/buildguy/project/engine/src/org/pentaho/di/trans/steps/ldapinput/LdapProtocolFactory.java:92: error: cannot find symbol
Wed Mar 01 13:35:56 EST 2017 - stdout - head:     [javac]     String connectionType = environmentSubstitute( meta.getProtocol() );
Wed Mar 01 13:35:56 EST 2017 - stdout - head:     [javac]                             ^
Wed Mar 01 13:35:56 EST 2017 - stdout - head:     [javac]   symbol:   method environmentSubstitute(String)
Wed Mar 01 13:35:56 EST 2017 - stdout - head:     [javac]   location: class LdapProtocolFactory